.github/workflows/pr.yaml

```diff
@@ -26,9 +26,47 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check-image:
+    name: Check Image & Changes
+    runs-on: ubuntu-latest
+    outputs:
+      relevant_changes: ${{ steps.check-image.outputs.relevant_changes }}
+      image_exists: ${{ steps.check-image.outputs.image_exists }}
+    steps:
+      - name: Run image & changes check
+        uses: ./.github/actions/check-image-and-changes
+
+  linting:
+    name: Run Linting & Prettier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        uses: backstage/actions/yarn-install@v0.6.17
+        with:
+          cache-prefix: ${{ runner.os }}-v22
+
+      - name: Run Prettier
+        run: yarn prettier:check --continue --affected
+
+      - name: Run Lint
+        run: yarn run lint:check --continue --affected
+
   build:
     name: Build with Node.js ${{ matrix.node-version }}
     runs-on: ubuntu-latest
+    needs: check-image
+    if: ${{ needs.check-image.outputs.relevant_changes == 'true' || needs.check-image.outputs.image_exists == 'false' }}
```

Comment:
looks like the build step will be completely skipped here. can we do a similar check as you have in pr-build-image.yaml line 61? https://github.com/redhat-developer/rhdh/pull/2575/files#diff-2b0fff9a121c4bf67a04b4a82b44d2f2a5cc48a69d8c299b46072978d7cf1212R61

Reply:
@nickboldt  You're right! The current logic would completely skip the build step if no relevant changes are detected and the image already exists.

I have updated the workflow to ensure that the job always runs (if: always()) and added an explicit check inside the first step. If the conditions indicate that no build is needed, the job exits early with exit 0, similar to what we did in pr-build-image.yaml at line 61.

This way, the workflow structure remains intact, and we avoid unnecessary builds while still ensuring required jobs are executed properly.

Let me know if you have any concerns! Thanks for the feedback.